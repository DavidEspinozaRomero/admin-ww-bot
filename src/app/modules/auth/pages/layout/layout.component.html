<svg
  class="bg-svg-position"
  xmlns="http://www.w3.org/2000/svg"
  version="1.1"
  xmlns:xlink="http://www.w3.org/1999/xlink"
  xmlns:svgjs="http://svgjs.com/svgjs"
  width="100vw"
  height="100vh"
  preserveAspectRatio="none"
  viewBox="0 0 1440 560"
>
  <g mask='url("#SvgjsMask1004")' fill="none">
    <rect width="1440" height="560" x="0" y="0" fill="#0e2a47"></rect>
    <path
      d="M14.64 -53.62L79.59 -16.12L79.59 58.88L14.64 96.38L-50.31 58.88L-50.31 -16.12zM14.64 171.38L79.59 208.88L79.59 283.88L14.64 321.38L-50.31 283.88L-50.31 208.88zM79.59 283.88L144.55 321.38L144.55 396.38L79.59 433.88L14.64 396.38L14.64 321.38zM144.55 396.38L209.5 433.88L209.5 508.88L144.55 546.38L79.59 508.88L79.59 433.88zM209.5 508.88L274.45 546.38L274.45 621.38L209.5 658.88L144.55 621.38L144.55 546.38zM274.45 -53.62L339.41 -16.12L339.41 58.88L274.45 96.38L209.5 58.88L209.5 -16.12zM339.41 283.88L404.36 321.38L404.36 396.38L339.41 433.88L274.45 396.38L274.45 321.38zM339.41 508.88L404.36 546.38L404.36 621.38L339.41 658.88L274.45 621.38L274.45 546.38zM469.32 58.88L534.27 96.38L534.27 171.38L469.32 208.88L404.36 171.38L404.36 96.38zM469.32 283.88L534.27 321.38L534.27 396.38L469.32 433.88L404.36 396.38L404.36 321.38zM404.36 396.38L469.32 433.88L469.32 508.88L404.36 546.38L339.41 508.88L339.41 433.88zM534.27 -53.62L599.22 -16.12L599.22 58.88L534.27 96.38L469.32 58.88L469.32 -16.12zM534.27 396.38L599.22 433.88L599.22 508.88L534.27 546.38L469.32 508.88L469.32 433.88zM664.18 -53.62L729.13 -16.12L729.13 58.88L664.18 96.38L599.22 58.88L599.22 -16.12zM729.13 58.88L794.09 96.38L794.09 171.38L729.13 208.88L664.18 171.38L664.18 96.38zM729.13 508.88L794.09 546.38L794.09 621.38L729.13 658.88L664.18 621.38L664.18 546.38zM923.99 -53.62L988.95 -16.12L988.95 58.88L923.99 96.38L859.04 58.88L859.04 -16.12zM988.95 58.88L1053.9 96.38L1053.9 171.38L988.95 208.88L923.99 171.38L923.99 96.38zM988.95 283.88L1053.9 321.38L1053.9 396.38L988.95 433.88L923.99 396.38L923.99 321.38zM923.99 396.38L988.95 433.88L988.95 508.88L923.99 546.38L859.04 508.88L859.04 433.88zM1053.9 -53.62L1118.85 -16.12L1118.85 58.88L1053.9 96.38L988.95 58.88L988.95 -16.12zM1053.9 396.38L1118.85 433.88L1118.85 508.88L1053.9 546.38L988.95 508.88L988.95 433.88zM1248.76 58.88L1313.71 96.38L1313.71 171.38L1248.76 208.88L1183.81 171.38L1183.81 96.38zM1183.81 171.38L1248.76 208.88L1248.76 283.88L1183.81 321.38L1118.85 283.88L1118.85 208.88zM1378.67 283.88L1443.62 321.38L1443.62 396.38L1378.67 433.88L1313.72 396.38L1313.72 321.38zM1378.67 508.88L1443.62 546.38L1443.62 621.38L1378.67 658.88L1313.72 621.38L1313.72 546.38zM1443.62 -53.62L1508.58 -16.12L1508.58 58.88L1443.62 96.38L1378.67 58.88L1378.67 -16.12z"
      stroke="#03305d"
      stroke-width="2"
    ></path>
    <path
      d="M7.14 -53.62 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM72.09 -16.12 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM72.09 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM7.14 96.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM-57.81 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM-57.81 -16.12 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM7.14 171.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM72.09 208.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM72.09 283.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM7.14 321.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM-57.81 283.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM-57.81 208.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM137.05 321.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM137.05 396.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM72.09 433.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM7.14 396.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM202 433.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM202 508.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM137.05 546.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM72.09 508.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM266.95 546.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM266.95 621.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM202 658.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM137.05 621.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM266.95 -53.62 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM331.91 -16.12 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM331.91 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM266.95 96.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM202 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM202 -16.12 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM331.91 283.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM396.86 321.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM396.86 396.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM331.91 433.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM266.95 396.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM266.95 321.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM331.91 508.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM396.86 546.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM396.86 621.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM331.91 658.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM461.82 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM526.77 96.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM526.77 171.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM461.82 208.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM396.86 171.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM396.86 96.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM461.82 283.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM526.77 321.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM526.77 396.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM461.82 433.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM461.82 508.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM526.77 -53.62 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM591.72 -16.12 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM591.72 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM461.82 -16.12 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM591.72 433.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM591.72 508.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM526.77 546.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM656.68 -53.62 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM721.63 -16.12 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM721.63 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM656.68 96.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM786.59 96.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM786.59 171.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM721.63 208.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM656.68 171.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM721.63 508.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM786.59 546.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM786.59 621.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM721.63 658.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM656.68 621.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM656.68 546.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM916.49 -53.62 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM981.45 -16.12 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM981.45 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM916.49 96.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM851.54 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM851.54 -16.12 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1046.4 96.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1046.4 171.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM981.45 208.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM916.49 171.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM981.45 283.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1046.4 321.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1046.4 396.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM981.45 433.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM916.49 396.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM916.49 321.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM981.45 508.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM916.49 546.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM851.54 508.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM851.54 433.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1046.4 -53.62 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1111.35 -16.12 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1111.35 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1111.35 433.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1111.35 508.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1046.4 546.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1241.26 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1306.21 96.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1306.21 171.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1241.26 208.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1176.31 171.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1176.31 96.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1241.26 283.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1176.31 321.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1111.35 283.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1111.35 208.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1371.17 283.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1436.12 321.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1436.12 396.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1371.17 433.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1306.22 396.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1306.22 321.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1371.17 508.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1436.12 546.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1436.12 621.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1371.17 658.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1306.22 621.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1306.22 546.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1436.12 -53.62 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1501.08 -16.12 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1501.08 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1436.12 96.38 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1371.17 58.88 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0zM1371.17 -16.12 a7.5 7.5 0 1 0 15 0 a7.5 7.5 0 1 0 -15 0z"
      fill="#03305d"
    ></path>
    <path
      d="M74.4 184.72L117.71 209.72L117.71 259.72L74.4 284.72L31.1 259.72L31.1 209.72zM31.1 259.72L74.4 284.72L74.4 334.72L31.1 359.72L-12.2 334.72L-12.2 284.72zM31.1 409.72L74.4 434.72L74.4 484.72L31.1 509.72L-12.2 484.72L-12.2 434.72zM117.71 409.72L161.01 434.72L161.01 484.72L117.71 509.72L74.4 484.72L74.4 434.72zM161.01 484.72L204.31 509.72L204.31 559.72L161.01 584.72L117.71 559.72L117.71 509.72zM247.61 184.72L290.92 209.72L290.92 259.72L247.61 284.72L204.31 259.72L204.31 209.72zM290.92 -40.28L334.22 -15.28L334.22 34.72L290.92 59.72L247.61 34.72L247.61 -15.28zM334.22 184.72L377.52 209.72L377.52 259.72L334.22 284.72L290.92 259.72L290.92 209.72zM420.82 184.72L464.13 209.72L464.13 259.72L420.82 284.72L377.52 259.72L377.52 209.72zM464.13 109.72L507.43 134.72L507.43 184.72L464.13 209.72L420.82 184.72L420.82 134.72zM507.43 184.72L550.73 209.72L550.73 259.72L507.43 284.72L464.13 259.72L464.13 209.72zM464.13 259.72L507.43 284.72L507.43 334.72L464.13 359.72L420.82 334.72L420.82 284.72zM507.43 484.72L550.73 509.72L550.73 559.72L507.43 584.72L464.13 559.72L464.13 509.72zM594.03 334.72L637.34 359.72L637.34 409.72L594.03 434.72L550.73 409.72L550.73 359.72zM637.34 -40.28L680.64 -15.28L680.64 34.72L637.34 59.72L594.03 34.72L594.03 -15.28zM680.64 184.72L723.94 209.72L723.94 259.72L680.64 284.72L637.34 259.72L637.34 209.72zM680.64 484.72L723.94 509.72L723.94 559.72L680.64 584.72L637.34 559.72L637.34 509.72zM767.24 334.72L810.55 359.72L810.55 409.72L767.24 434.72L723.94 409.72L723.94 359.72zM810.55 259.72L853.85 284.72L853.85 334.72L810.55 359.72L767.24 334.72L767.24 284.72zM810.55 409.72L853.85 434.72L853.85 484.72L810.55 509.72L767.24 484.72L767.24 434.72zM940.45 34.72L983.76 59.72L983.76 109.72L940.45 134.72L897.15 109.72L897.15 59.72zM897.15 109.72L940.45 134.72L940.45 184.72L897.15 209.72L853.85 184.72L853.85 134.72zM897.15 409.72L940.45 434.72L940.45 484.72L897.15 509.72L853.85 484.72L853.85 434.72zM983.76 109.72L1027.06 134.72L1027.06 184.72L983.76 209.72L940.45 184.72L940.45 134.72zM1027.06 334.72L1070.36 359.72L1070.36 409.72L1027.06 434.72L983.76 409.72L983.76 359.72zM1070.36 -40.28L1113.66 -15.28L1113.66 34.72L1070.36 59.72L1027.06 34.72L1027.06 -15.28zM1113.66 484.72L1156.96 509.72L1156.96 559.72L1113.66 584.72L1070.36 559.72L1070.36 509.72zM1200.27 34.72L1243.57 59.72L1243.57 109.72L1200.27 134.72L1156.96 109.72L1156.96 59.72zM1200.27 184.72L1243.57 209.72L1243.57 259.72L1200.27 284.72L1156.96 259.72L1156.96 209.72zM1156.96 259.72L1200.27 284.72L1200.27 334.72L1156.96 359.72L1113.66 334.72L1113.66 284.72zM1286.87 34.72L1330.17 59.72L1330.17 109.72L1286.87 134.72L1243.57 109.72L1243.57 59.72zM1286.87 334.72L1330.17 359.72L1330.17 409.72L1286.87 434.72L1243.57 409.72L1243.57 359.72zM1373.48 184.72L1416.78 209.72L1416.78 259.72L1373.48 284.72L1330.17 259.72L1330.17 209.72zM1373.48 334.72L1416.78 359.72L1416.78 409.72L1373.48 434.72L1330.17 409.72L1330.17 359.72zM1416.78 109.72L1460.08 134.72L1460.08 184.72L1416.78 209.72L1373.48 184.72L1373.48 134.72zM1416.78 259.72L1460.08 284.72L1460.08 334.72L1416.78 359.72L1373.48 334.72L1373.48 284.72zM1460.08 484.72L1503.38 509.72L1503.38 559.72L1460.08 584.72L1416.78 559.72L1416.78 509.72z"
      stroke="rgba(6, 80, 156, 0.93)"
      stroke-width="2"
    ></path>
  </g>
  <defs>
    <mask id="SvgjsMask1004">
      <rect width="1440" height="560" fill="#ffffff"></rect>
    </mask>
  </defs>
</svg>
<router-outlet ></router-outlet>
