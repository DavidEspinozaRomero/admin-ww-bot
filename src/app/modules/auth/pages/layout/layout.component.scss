.bg-svg-position {
  position: absolute;
  z-index: -1;
}