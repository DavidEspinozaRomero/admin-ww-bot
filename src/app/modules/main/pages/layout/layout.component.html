<nav class="sidebar " [ngClass]="{ menu_extend: !colapse }">
  <p
    class="mt-4 mb-3 ps-3"
    role="button"
    style="inline-size: max-content"
    (click)="colapse = !colapse"
  >
    <i class="fs-5 bi bi-box-arrow-left"></i> <span class="ms-3">BRAND</span>
  </p>
  <ul class="nav d-block h-100">
    <li
      *ngFor="let item of menu"
      class="nav-item"
      role="button"
      [routerLink]="item.route"
      [routerLinkActive]="'hovered'"
    >
      <a class="nav-link">
        <i class="fs-5" [class]="item.icon"></i>
        <span class="ms-3"> {{ item.label | titlecase }} </span>
      </a>
    </li>

    <li class="nav-item mt-auto">
      <a class="nav-link" (click)="logout()">
        <i class="fs-5 bi bi-box-arrow-left"></i>
        <span class="ms-3">Log out</span>
      </a>
    </li>
  </ul>
</nav>

<main class="main p-3" [ngClass]="{ menu_colapse: !colapse }">
  <router-outlet></router-outlet>
</main>
