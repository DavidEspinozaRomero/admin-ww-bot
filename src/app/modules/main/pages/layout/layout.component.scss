@import "./src/assets/scss/colors";

/* The sidebar menu */
.sidebar {
  height: 100%; /* 100% Full-height */
  width: 4rem; /* 0 width - change this with JavaScript */
  position: fixed; /* Stay in place */
  z-index: 1; /* Stay on top */
  top: 0;
  left: 0;
  background-color: $bg-blue; /* Black*/
  overflow-x: hidden; /* Disable horizontal scroll */
  padding-top: 60px; /* Place content 60px from the top */
  transition: 0.5s; /* 0.5 second transition effect to slide in the sidebar */
  border-left: 1rem solid $bg-blue;
}

/* The sidebar links */
.sidebar li {
  display: block;
  transition: 0.3s;
  border-radius: 1rem 0 0 1rem;
}

/* When you mouse over the navigation links, change their color */
.sidebar li:hover {
  // color: $bg-blue;
  background-color: $bg-white;
}

/* Position and style the close button (top right corner) */
.sidebar .closebtn {
  position: absolute;
  top: 0;
  right: 25px;
  font-size: 36px;
  margin-left: 50px;
}

.menu_extend {
  width: 10rem;
}

.menu_colapse {
  margin-left: 10rem;
}

/* On smaller screens, where height is less than 450px, change the style of the sidenav (less padding and li smaller font size) */
@media screen and (max-height: 450px) {
  .sidebar {
    padding-top: 15px;
  }
  .sidebar a {
    font-size: 18px;
  }
}

// .navidation {
//   width: 10rem;
//   height: 100%;
//   border-left: 1rem solid $blue;
//   // transition: 300ms ease-in-out;
//   overflow: hidden;
//   ul {
//     position: absolute;
//     top: 0;
//     left: 0;
//     width: 100%;
//   }
//   li {
//     position: relative;
//   }
//   .hovered a,
//   li a {
//     color: white;
//     background-color: $blue;
//     border-radius: 1rem 0 0 1rem;
//   }
//   li:nth-child(1) {
//     margin-top: 1rem;
//     margin-bottom: 3rem;
//   }
//   li:last-child {
//     margin-top: auto;
//     color: red !important;
//   }
//   .hovered a,
//   li:hover a {
//     background-color: white;
//     border-radius: 1rem 0 0 1rem;
//     color: $blue;
//   }

//   li:hover a::after,
//   .hovered a::after {
//     content: "";
//     position: absolute;
//     right: 0;
//     top: -1rem;
//     width: 1rem;
//     height: 1rem;
//     background-color: transparent;
//     border-radius: 1rem;
//     box-shadow: 1rem 1rem 0 0.5rem white;
//   }
//   li:hover a::before,
//   .hovered a::before {
//     content: "";
//     position: absolute;
//     right: 0;
//     bottom: -1rem;
//     width: 1rem;
//     height: 1rem;
//     background-color: transparent;
//     border-radius: 1rem;
//     box-shadow: 1rem -1rem 0 0.5rem white;
//     z-index: 1;
//   }
// }

// main {
//   position: absolute;
//   left: 10rem;
//   width: calc(100% - 10rem);
//   transition: 300ms ease-in-out;
// }
// .extend {
//   left: 4rem;
//   width: calc(100% - 4rem);
// }

// .colapse {
//   width: 4rem;
// }
