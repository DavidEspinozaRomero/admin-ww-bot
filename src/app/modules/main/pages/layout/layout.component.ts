import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-layout',
  templateUrl: './layout.component.html',
  styleUrls: ['./layout.component.scss'],
})
export class LayoutComponent implements OnInit {
  //#region variables
  menu = [
    // { icon: '', label: '', route: '' },
    { icon: 'bi bi-house', label: 'dashboard', route: './dashboard' },
    { icon: 'bi bi-house', label: 'settings', route: './settings' },
  ];
  //#endregion variables
  constructor() {}

  ngOnInit(): void {}
}
