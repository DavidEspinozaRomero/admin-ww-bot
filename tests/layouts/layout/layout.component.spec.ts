import { ComponentFixture } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { screen, render, RenderResult } from '@testing-library/angular';
import { LayoutComponent } from '../../../src/app/layouts/layout/layout.component';

//#region Mocks
const menuMock = [
  // { icon: '', label: '', route: '' },
  // { icon: 'bi bi-building', label: 'brand name', route: './brand' },
  { icon: 'bi bi-house-door', label: 'dashboard', route: '/dashboard' },
  { icon: 'bi bi-person', label: 'contacts', route: '/contacts' },
  { icon: 'bi bi-chat', label: 'messages', route: '/messages' },
  { icon: 'bi bi-hourglass', label: 'history', route: '/history' },
  { icon: 'bi bi-gear', label: 'settings', route: '/settings' },
  { icon: 'bi bi-question-circle', label: 'help', route: '/help' },
  // { icon: 'bi bi-box-arrow-left', label: 'log out' },
];
//#endregion Mocks

describe('LayoutComponent', () => {
  let rendered: RenderResult<LayoutComponent>;
  let fixture: ComponentFixture<LayoutComponent>;
  let component: LayoutComponent;
  let compiled: HTMLElement;

  beforeEach(async () => {
    rendered = await render(LayoutComponent, {
      imports: [RouterTestingModule],
    });
  });
  beforeEach(() => {
    fixture = rendered.fixture;
    component = rendered.fixture.componentInstance;
    compiled = rendered.fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  describe('Layout', () => {
    it('should create', () => {
      expect(compiled).toMatchSnapshot();
    });
    it('has a sidebar menu', () => {
      const sidebar = screen.getByRole('navigation', { name: 'sidebar' });
      expect(sidebar).toBeInTheDocument();
    });
    it('has inside the sidebar: brand, nav-items, logout', () => {
      const brand = screen.getByRole('button', {
        name: 'brand and sidebar toggle btn',
      });

      const dashboard = screen.getByText(/dashboard/i);
      const contacts = screen.getByText(/Contacts/i);
      const messages = screen.getByText(/Messages/i);
      const history = screen.getByText(/History/i);
      const settings = screen.getByText(/Settings/i);
      const help = screen.getByText(/Help/i);
      const logout = screen.getByText('Log out');

      expect(brand).toBeInTheDocument();
      expect(dashboard).toBeInTheDocument();
      expect(contacts).toBeInTheDocument();
      expect(messages).toBeInTheDocument();
      expect(history).toBeInTheDocument();
      expect(settings).toBeInTheDocument();
      expect(help).toBeInTheDocument();
      expect(logout).toBeInTheDocument();
    });
  });
  describe('Methods', () => {
    it('has logout', () => {
      expect(typeof component.logout).toBe('function');
    });
  });
});
