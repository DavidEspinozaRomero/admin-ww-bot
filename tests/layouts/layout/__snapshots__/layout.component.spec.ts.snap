// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutComponent Layout should create 1`] = `
<div
  id="root1"
>
  <nav
    aria-label="sidebar"
    class="sidebar"
  >
    <p
      aria-label="brand and sidebar toggle btn"
      class="my-3 ps-3 text-white"
      role="button"
      style="inline-size: max-content;"
    >
      <i
        class="fs-5 bi bi-apple"
      />
      <span
        class="ms-3"
      >
        BRAND
      </span>
    </p>
    <ul
      class="nav d-block h-100"
    >
      <li
        class="nav-item"
        role="button"
        tabindex="0"
      >
        <a
          class="nav-link"
        >
          <i
            class="fs-5 bi bi-house-door"
          />
          <span
            class="ms-3"
          >
             Dashboard 
          </span>
        </a>
      </li>
      <li
        class="nav-item"
        role="button"
        tabindex="0"
      >
        <a
          class="nav-link"
        >
          <i
            class="fs-5 bi bi-person"
          />
          <span
            class="ms-3"
          >
             Contacts 
          </span>
        </a>
      </li>
      <li
        class="nav-item"
        role="button"
        tabindex="0"
      >
        <a
          class="nav-link"
        >
          <i
            class="fs-5 bi bi-chat"
          />
          <span
            class="ms-3"
          >
             Messages 
          </span>
        </a>
      </li>
      <li
        class="nav-item"
        role="button"
        tabindex="0"
      >
        <a
          class="nav-link"
        >
          <i
            class="fs-5 bi bi-hourglass"
          />
          <span
            class="ms-3"
          >
             History 
          </span>
        </a>
      </li>
      <li
        class="nav-item"
        role="button"
        tabindex="0"
      >
        <a
          class="nav-link"
        >
          <i
            class="fs-5 bi bi-gear"
          />
          <span
            class="ms-3"
          >
             Settings 
          </span>
        </a>
      </li>
      <li
        class="nav-item"
        role="button"
        tabindex="0"
      >
        <a
          class="nav-link"
        >
          <i
            class="fs-5 bi bi-question-circle"
          />
          <span
            class="ms-3"
          >
             Help 
          </span>
        </a>
      </li>
      
      <li
        class="nav-item mt-auto"
      >
        <a
          class="nav-link"
        >
          <i
            class="fs-5 bi bi-box-arrow-left"
          />
          <span
            class="ms-3"
          >
            Log out
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <main
    class="main p-3 vh-100"
  >
    <router-outlet />
    
  </main>
</div>
`;
